doc: matter: Add documentation for last fabric removal (#12558)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst

* `Matter Simple Setup for Wi-Fi Overview`_
* `Matter Simple Setup for Thread Overview`_

Reaction to the last Matter fabric removal
==========================================

When a Matter device joins the Matter network it starts belonging to a specific Matter fabric.
To ensure interoperability of multiple applications and ecosystems the device can join multiple Matter fabrics created by multiple ecosystems using the multi-fabric feature.
As a result, the device can belong to multiple fabrics and can be managed by multiple Matter controllers.
To remove the device from a specific fabric a Matter controller sends the leave request to it and then removes all information collected from the device.
The device receives the leave request and removes all information about the leaving fabric.
When the device leaves the last fabric there are several reactions to that to be set.

> To enable one of the reactions to the last fabric removal choose and set the following configuration option:

Reviewer: Could we write somewhere what is the default behavior?

* `Matter Simple Setup for Wi-Fi Overview`_
* `Matter Simple Setup for Thread Overview`_

Reaction to the last Matter fabric removal
==========================================

> When a Matter device is commissioned by the Matter controller it starts belonging to a specific Matter fabric.

Reviewer 2: ```suggestion
When a Matter device is commissioned by the Matter controller, it then belongs to a specific Matter fabric.
```
This section is worded slightly different from the similar section on the other page. Check that they don't contradict each other.

You could also consider using an include from one page to the other.

Author: I've tried to include that part. Did I do it correctly? :)

> To ensure interoperability of multiple applications and ecosystems the device can join multiple Matter fabrics created by multiple ecosystems using the multi-fabric feature.

Reviewer 2: ```suggestion
To ensure the interoperability of multiple applications and ecosystems, the device can join the Matter fabrics created by multiple ecosystems using the multi-fabric feature.
```

As a result, the device can belong to multiple fabrics and can be managed by multiple Matter controllers.

> To remove the device from a specific fabric a Matter controller sends the leave request to it and then removes all information collected from the device.

Reviewer 2: ```suggestion

To remove the device from a fabric, a Matter controller sends the leave request to it and then removes all information collected from the device.
```

> The device receives the leave request and removes all information about the leaving fabric.

Reviewer 2: ```suggestion
The device receives the leave request and removes all information about the fabric it is leaving.
```

> When the device leaves the last fabric there are several reactions to that to be set.

Reviewer 2: ```suggestion
When the device leaves the last fabric, one of several reactions can be set to happen.
```

> To enable one of the reactions to the last fabric removal choose and set the following configuration option:

Reviewer 2: ```suggestion
To enable one of the reactions to the last fabric removal, set the corresponding Kconfig option to `` y``:
```

> * :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_NONE` to ``y`` to do not react to the last fabric removal.
>   The device will leave all saved data and network credentials and not reboot.

Reviewer 2: ```suggestion
* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_NONE` - Do not react to the last fabric removal.
  The device will keep all saved data and network credentials, and will not reboot.
```

> * :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_ONLY` to ``y`` to remove all saved network credentials.
>   The device will remove all saved network credentials and leave application-specific non-volatile data.

Reviewer 2: ```suggestion
* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_ONLY` - Remove all saved network credentials.
  The device will remove all saved network credentials, keep application-specific non-volatile data, and will not reboot.
```

> * :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_PAIRING_START` to ``y`` to remove all saved network credentials and start Bluetooth LE advertising.
>   The device will remove all saved network credentials, leave application-specific non-volatile data, and start advertising Bluetooth LE Matter service.
>   After that, it will be ready for the new commissioning to Matter over Bluetooth LE.

Reviewer 2: ```suggestion
* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_PAIRING_START` - Remove all saved network credentials and start Bluetooth LE advertising.
  The device will remove all saved network credentials, keep application-specific non-volatile data, and start advertising Bluetooth LE Matter service.
  After that, it will be ready for commissioning to Matter over Bluetooth LE.
```

> * :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` to ``y`` to remove all saved network credentials and reboot the device.
>   The device will remove all saved network credentials and perform a reboot.
>   After that, it will be restored to the factory settings.
>   *This option is selected by default*.

Reviewer 2: ```suggestion
* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` - Remove all saved network credentials and reboot the device.
  The device will remove all saved network credentials and then reboot.
  After that, it will be restored to the factory settings.
  
  This option is selected by default.
```

> Additionally, set the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` to ``y`` to remove all non-volatile data stored on the device when the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` option is selected.

Reviewer 2: ```suggestion
when the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` Kconfig option is selected, set the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option to ``y`` to also remove all non-volatile data stored on the device.
```

.. include:: ../end_product/last_fabric_removal_delegate.rst
    :start-after: matter_last_fabric_removal_description_start
    :end-before: matter_last_fabric_removal_description_end

When the device leaves the last fabric, one of several reactions can be set to happen.

To enable one of the reactions to the last fabric removal, set the corresponding Kconfig option to ``y``:

* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_NONE` - Do not react to the last fabric removal.
  The device will keep all saved data and network credentials, and will not reboot.
* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_ONLY` - Remove all saved network credentials.
  The device will remove all saved network credentials, keep application-specific non-volatile data, and will not reboot.
* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_PAIRING_START` - Remove all saved network credentials and start Bluetooth LE advertising.
  The device will remove all saved network credentials, keep application-specific non-volatile data, and start advertising Bluetooth LE Matter service.
  After that, it will be ready for commissioning to Matter over Bluetooth LE.

> * :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` - Remove all saved network credentials and reboot the device.
>   The device will remove all saved network credentials and then reboot.
>   Moreover, if the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` is set to ``y`` the device will also remove all non-volatile data stored on the device including application-specific entries.
>   After that, it will be restored to the factory settings.
> 
>   This option is selected by default.

Reviewer 2: ```suggestion
* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` - Remove all saved network credentials and reboot the device.
  This option is selected by default.
  
  When the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option is also set to ``y``, the device will also remove all non-volatile data stored on the device, including application-specific entries.
  This means the device is restored to the factory settings. 
```

.. note::
  The :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option is set to ``y`` by default.

>   To disable removing application-specific non-volatile data when the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` Kconfig option is selected set the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option to ``n``.

Reviewer 2: ```suggestion
  To disable removing application-specific non-volatile data when the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` Kconfig option is selected, set the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option to ``n``.
```
